src/test/java/de/unijena/cheminf/clustering/desccalc/DescriptorTest.java

```diff
@@ -55,214 +63,4106 @@ public void test_MOLECULAR_WEIGHT() throws Exception {
+            Assertions.assertNotEquals(0f, tmpMatrix[0][2]);
+            Assertions.assertNotEquals(0f, tmpMatrix[0][3]);
+            Assertions.assertNotEquals(0f, tmpMatrix[0][4]);
+
+            tmpMatrix = new float[][]
+                    {
+                            {0f, 0f, 0f, 0f, 0f}
+                    };
+            aNanPositions = Collections.synchronizedList(new LinkedList<>());
+            Assertions.assertTrue(
+                    Descriptor.setDescriptorsForMoleculesByDescriptorParallelization(
+                            tmpDescriptors,
+                            tmpMoleculesArray,
+                            tmpMatrix,
+                            tmpStartIndex,
+                            tmpIsParallelCalculation,
+                            aNanPositions
+                    )
+            );
+            Assertions.assertNotEquals(0f, tmpMatrix[0][0]);
+            Assertions.assertNotEquals(0f, tmpMatrix[0][1]);
+            Assertions.assertNotEquals(0f, tmpMatrix[0][2]);
+            Assertions.assertNotEquals(0f, tmpMatrix[0][3]);
+            Assertions.assertNotEquals(0f, tmpMatrix[0][4]);
+        } catch (Exception anException) {
+            Assertions.fail();
+        }
+    }
+
+    /**
+     * Tests method for descriptor AUTOCORRELATION_POLARIZABILITY
+     * No validated result because the descriptor itself is not validated in the CDK
+     * Result can be printed to see if descriptor calculates values
+     */
+    @Test
+    public void test_AUTOCORRELATION_POLARIZABILITY() throws Exception {
+        String tmpSmiles = "Clc1ccccc1";
+        SmilesParser tmpSmilesParser = new SmilesParser(SilentChemObjectBuilder.getInstance());
+        IAtomContainer tmpMolecule = tmpSmilesParser.parseSmiles(tmpSmiles);
+        IAtomContainer[] tmpMoleculesArray = new IAtomContainer[]{tmpMolecule};
+        int tmpStartIndex = 0;
+        Descriptor[] tmpDescriptors = new Descriptor[]{Descriptor.AUTOCORRELATION_POLARIZABILITY};
+        boolean tmpIsParallelCalculation = false;
+
+        try {
+            Assertions.assertEquals(5, Descriptor.getNumberOfComponents(tmpDescriptors));
+
+            float[][] tmpMatrix = new float[][]
+                    {
+                            {0f, 0f, 0f, 0f, 0f}
+                    };
+            List<int[]> aNanPositions = Collections.synchronizedList(new LinkedList<>());
+            Assertions.assertTrue(
+                    Descriptor.setDescriptorsForMoleculesByMoleculeParallelizationSynchronized(
+                            tmpDescriptors,
+                            tmpMoleculesArray,
+                            tmpMatrix,
+                            tmpStartIndex,
+                            tmpIsParallelCalculation,
+                            aNanPositions
+                    )
+            );
+            //System.out.println("Synchronized calculation results:");
+            //System.out.println(Arrays.toString(tmpMatrix[0]));
```

Comment:
Maybe at least test whether the results are the same for every parallelization method?